### PR TITLE
Add Venue Block with support for search or create

### PR DESCRIPTION
**Pull request by mitogh**

### CODE REVIEW

File: src/Tribe/Blocks/Event_Venue.php [hunk 1]
```diff
@@ -30,4 +30,51 @@ public function render( $attributes = array() ) {
...
-}
\ No newline at end of file
+
+	/**
+	 * Attach hooks associated with this block
+	 *
+	 * @return mixed|void
+	 */
+	public function hook() {
+		if ( class_exists( 'Tribe__Events__Main' ) ) {
```
Comment (bordoni) [inline src/Tribe/Blocks/Event_Venue.php]: Invert this conditional and return.

File: src/Tribe/Blocks/Event_Venue.php [hunk 2]
```diff
@@ -30,4 +30,51 @@ public function render( $attributes = array() ) {
...
+	 * Remove venues associated to an event that are no longer relevant
+	 *
+	 * @since TBD
+	 *
+	 * @param $post_id
+	 */
+	public function clear_venues( $post_id ) {
+		$status = get_post_status( $post_id );
+		if ( 'publish' !== $status ) {
+			return;
```
Comment (bordoni) [inline src/Tribe/Blocks/Event_Venue.php]: Lets return `false`.

File: src/Tribe/Blocks/Event_Venue.php [hunk 3]
```diff
@@ -30,4 +30,51 @@ public function render( $attributes = array() ) {
...
+		}
+
+		$key = '_EventTempVenues';
+		$venues = get_post_meta( $post_id, $key );
+
+		if ( empty( $venues ) || ! is_array( $venues ) ) {
+			return;
```
Comment (bordoni) [inline src/Tribe/Blocks/Event_Venue.php]: Lets return `false`.

File: src/Tribe/Blocks/Event_Venue.php [hunk 4]
```diff
@@ -30,4 +30,51 @@ public function render( $attributes = array() ) {
...
+			if ( $current_venue === absint( $venue_id ) ) {
+				continue;
+			}
+
+			if ( 'draft' === get_post_status( $venue_id ) ) {
+				wp_delete_post( $venue_id );
+			}
+		}
+
+		delete_post_meta( $post_id, $key );
```
Comment (bordoni) [inline src/Tribe/Blocks/Event_Venue.php]: Lets return a status of if we deleted or not.

**Comment by mitogh** [comment]
Changes has been applied @bordoni

**Pull request merged**